core/src/main/scala/org/scalacheck/ops/GenOrThrow.scala

```diff
@@ -68,10 +68,30 @@ class GenOrThrow[T: ClassTag](val gen: Gen[T]) {
    * @throws EmptyGenSampleException when samples cannot be generated after a default number of attempts
    */
   @deprecated("The meaning of this method will change in the next major version. " +
-    "If you need a random sample on each call, you should use .randomOrThrow() instead",
+    "If you need a random sample on each call, you should use .getOrThrowPure or .randomOrThrow() instead",
     "2.3.0")
   def getOrThrow: T = getOrThrow(GenConfig.default.withSeed(Seed.random()))
 
+  /**
+    * Get the next instance of this generate with the given configuration.
+    *
+    * @note this is a stateless function and calling this method with
+    *       the same config will always produce the same result.
+    *
+    * @note This is the syntax that will replace the parameterless
+    *       [[getOrThrow]] in the next major version.
+    */
+  def getOrThrowPure(implicit c: GenConfig): T = getOrThrow(c)
```

Comment:
Q: Why does this not have the default value like `randomOrThrow`?
Q: Why does this not have an empty `()` but `randomThrow()` does? It seems mildly inconsistent.

Reply:
> Q: Why does this not have the default value like `randomOrThrow`?

I made a mistake by using default implicit arguments before. I am planning to remove them on the next major version. In the meantime, I decided not to continue the bad pattern. I can add it for consistency, but I don't think it will change the behavior, since the default argument is defined on the companion object and is available without import everywhere.

> Q: Why does this not have an empty `()` but `randomThrow()` does? It seems mildly inconsistent.

According to the Scala style-guide (https://docs.scala-lang.org/style/method-invocation.html#arity-0):

> Scala allows the omission of parentheses on methods of arity-0 (no arguments). However, this syntax should only be used when the method in question has no side-effects (purely-functional). In other words, it would be acceptable to omit parentheses when calling `queue.size`, but not when calling `println()`... Religiously observing this convention will dramatically improve code readability and will make it much easier to understand at a glance the most basic operation of any given method. Resist the urge to omit parentheses simply to save two characters!

`randomOrThrow()` mutates the global random number generator and, thus, is not side-effect free. Each time you call it, you will get a different result. This is not the same as `.getOrThrowPure` which will always produce the same result from the same implicit (or explicit) config.